TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilderNodeListener.cs

```diff
@@ -0,0 +1,812 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.CodeModel;
+using TypeCobol.Compiler.Nodes;
+using TypeCobol.Compiler.Parser;
+
+namespace TypeCobol.Compiler.CupParser.NodeBuilder
+{
+    /// <summary>
+    /// A Node Listener that can also dispatch Program Class Builder actions
+    /// </summary>
+    public class ProgramClassBuilderNodeListener : NodeListener, IProgramClassBuilder
```

Comment:
This Listener interface redefines method `public virtual void OnNode(Node node, Program program)`.
Can we merge both interface?

Reply:
ProgramClassBuilderNodeListener is not an interface it implements NodeListener, it can be seen as an Adapter.